fdbserver/VersionedBTree.actor.cpp

```diff
@@ -1286,108 +1290,259 @@ int nextPowerOf2(uint32_t x) {
+			if (currLevel == 0) {
+				for (const auto& ER : L0PossibleEventReasonPairs) {
+					if (prevEvent != ER.first) {
+						result += "\n";
+						result += PagerEventsCodes[(size_t)ER.first];
+						result += "\n\t";
+						prevEvent = ER.first;
+					}
+					std::string num = std::to_string(eventReasons[(size_t)ER.first][(size_t)ER.second]);
+					result += PagerEventReasonsCodes[(size_t)ER.second];
+					result.append(16 - PagerEventReasonsCodes[(size_t)ER.second].length(), ' ');
+					result.append(8 - num.length(), ' ');
+					result += num;
+					result.append(13, ' ');
+				}
+			} else {
+				for (const auto& ER : possibleEventReasonPairs) {
+					if (prevEvent != ER.first) {
+						result += "\n";
+						result += PagerEventsCodes[(size_t)ER.first];
+						result += "\n\t";
+						prevEvent = ER.first;
+					}
+					std::string num = std::to_string(eventReasons[(size_t)ER.first][(size_t)ER.second]);
+					result += PagerEventReasonsCodes[(size_t)ER.second];
+					result.append(16 - PagerEventReasonsCodes[(size_t)ER.second].length(), ' ');
+					result.append(8 - num.length(), ' ');
+					result += num;
+					result.append(13, ' ');
+				}
+			}
+			return result;
+		}
+		void reportTrace(TraceEvent* t, int h) {
+			if (h == 0) {
+				for (const auto& ER : L0PossibleEventReasonPairs) {
+					t->detail(
+					    format(
+					        "L%d%s",
+					        h,
+					        (PagerEventsCodes[(size_t)ER.first] + PagerEventReasonsCodes[(size_t)ER.second]).c_str()),
+					    eventReasons[(size_t)ER.first][(size_t)ER.second]);
+				}
+			} else {
+				for (const auto& ER : possibleEventReasonPairs) {
+					t->detail(
+					    format(
+					        "L%d%s",
+					        h,
+					        (PagerEventsCodes[(size_t)ER.first] + PagerEventReasonsCodes[(size_t)ER.second]).c_str()),
+					    eventReasons[(size_t)ER.first][(size_t)ER.second]);
+				}
+			}
+		}
+	};
+	// Page levle events
 	struct Level {
-		unsigned int pageRead;
-		unsigned int pageReadExt;
-		unsigned int pageBuild;
-		unsigned int pageBuildExt;
-		unsigned int pageCommitStart;
-		unsigned int pageModify;
-		unsigned int pageModifyExt;
-		unsigned int lazyClearRequeue;
-		unsigned int lazyClearRequeueExt;
-		unsigned int lazyClearFree;
-		unsigned int lazyClearFreeExt;
-		unsigned int forceUpdate;
-		unsigned int detachChild;
-		double buildStoredPct;
-		double buildFillPct;
-		unsigned int buildItemCount;
-		double modifyStoredPct;
-		double modifyFillPct;
-		unsigned int modifyItemCount;
+		struct Counters {
+			unsigned int pageRead;
+			unsigned int pageReadExt;
+			unsigned int pageBuild;
+			unsigned int pageBuildExt;
+			unsigned int pageCommitStart;
+			unsigned int pageModify;
+			unsigned int pageModifyExt;
+			unsigned int lazyClearRequeue;
+			unsigned int lazyClearRequeueExt;
+			unsigned int lazyClearFree;
+			unsigned int lazyClearFreeExt;
+			unsigned int forceUpdate;
+			unsigned int detachChild;
+			EventReasonsArray eventReasons;
+		};
+		Counters metric;
```

Comment:
Since this struct contains many metrics, all of which are counters, it should be called `metrics` or `perhaps counters`.